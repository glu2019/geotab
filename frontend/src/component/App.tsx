import React, { useEffect, useState } from 'react'
import { connect } from 'react-redux'
import axios from 'axios'
import _ from 'lodash'
import { Spin, Button, notification } from 'antd'
import { LoadState } from '../store/index'
import { setLoading } from '../actions'
import Sudoku from './Sudoku'
import 'antd/dist/antd.css'
import './App.css'

export interface AppProps {
  loading: boolean
  setLoading: (loading: boolean) => {}
}

interface SelectedType {
  index: number
  value: number
}

const App: React.FunctionComponent<AppProps> = props => {
  const [selected, setSelected] = useState<Array<SelectedType>>([])
  const [puzzleArray, setPuzzleArray] = useState<Array<number>>(_.fill(Array(81), 0))

  function getSudoku() {
    props.setLoading(true)
    axios
      .get('/sudoku/board')
      .then(res => {
        setPuzzleArray(res.data)
        props.setLoading(false)
      })
      .catch(() => {
        notification.error({
          message: "Can't load the sudoku board, please try later!",
        })
      })
  }

  useEffect(() => {
    /* get initial board here */
    getSudoku()
  }, [])

  function onChange(i: number, value: number) {
    let newSelected = [...selected]
    const existIndex = newSelected.findIndex(e => e.index === i)
    if (existIndex !== -1) {
      newSelected = newSelected.filter(e => e.index !== i)
    } else {
      newSelected.push({ index: Number(i), value: Number(value) })
    }
    setSelected(newSelected)
  }

  function onRequest() {
    if (selected.length > 0) {
      const emptyArray = _.fill(Array(81), 0)
      selected.forEach(e => {
        emptyArray[e.index] = e.value
      })
      axios
        .post('/sudoku/board', {
          num: emptyArray,
        })
        .then(res => {
          setPuzzleArray(res.data)
        })
        .catch(() => {
          notification.error({
            message: "Can't load the sudoku board, please try later!",
          })
        })
    } else {
      getSudoku()
    }
  }

  return (
    <Spin spinning={props.loading} wrapperClassName="board-container">
      <h1>Sudoku Board</h1>
      <Sudoku puzzleArray={puzzleArray} onChange={onChange} selected={selected} />
      <div className="button-wrapper">
        <Button type="primary" size="large" onClick={onRequest}>
          Reload
        </Button>
      </div>
    </Spin>
  )
}

const mapStateToProps = (state: LoadState) => ({
  loading: state.loading,
})

const mapDispatchToProps = {
  setLoading,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)
