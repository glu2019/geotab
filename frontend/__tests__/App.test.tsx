import * as React from 'react'
import { mount, shallow, ShallowWrapper } from 'enzyme'
import { create, ReactTestRenderer } from 'react-test-renderer'
import App, { AppProps } from '../src/component/App'
import { withStore, axiosMock } from './test-utils'

let wrapper: ShallowWrapper<AppProps>
let snapshot: ReactTestRenderer

const apiRegex = /\/sudoku\/board/
const sudokuRes = [
  2,
  1,
  7,
  4,
  3,
  6,
  8,
  9,
  5,
  6,
  8,
  5,
  9,
  1,
  2,
  7,
  3,
  4,
  3,
  4,
  9,
  5,
  7,
  8,
  1,
  2,
  6,
  4,
  6,
  1,
  8,
  9,
  3,
  2,
  5,
  7,
  9,
  7,
  8,
  1,
  2,
  5,
  6,
  4,
  3,
  5,
  2,
  3,
  6,
  4,
  7,
  9,
  1,
  8,
  7,
  5,
  2,
  3,
  6,
  9,
  4,
  8,
  1,
  8,
  9,
  4,
  7,
  5,
  1,
  3,
  6,
  2,
  1,
  3,
  6,
  2,
  8,
  4,
  5,
  7,
  9,
]

beforeEach(() => {
  const app = <App />

  wrapper = shallow(withStore(app))
  snapshot = create(withStore(app))
})

beforeAll(() => {
  axiosMock.onGet(apiRegex).reply(async () => await [200, sudokuRes])
  axiosMock.onPost(apiRegex).reply(async () => await [200, sudokuRes])
})

afterAll(() => {
  axiosMock.onGet(apiRegex).passThrough()
  axiosMock.onPost(apiRegex).passThrough()
})

describe('<App />', () => {
  test('it matches the snapshot', () => {
    expect(snapshot).toMatchSnapshot()
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.cell')).not.toBeNull()
  })

  it('render board', () => {
    const mountWrapper = mount(withStore(<App />))
    expect(mountWrapper).toMatchSnapshot()
    mountWrapper
      .find('.cell')
      .at(0)
      .simulate('click')
    mountWrapper
      .find('.cell')
      .at(1)
      .simulate('click')
    expect(mountWrapper.find('.fix').length).toBe(2)
    mountWrapper
      .find('button')
      .at(0)
      .simulate('click')
    mountWrapper
      .find('.cell')
      .at(0)
      .simulate('click')
    expect(mountWrapper.find('.fix').length).toBe(1)
    mountWrapper
      .find('button')
      .at(1)
      .simulate('click')
    mountWrapper
      .find('button')
      .at(0)
      .simulate('click')
    expect(mountWrapper.find('.fix').length).toBe(0)
  })
})
