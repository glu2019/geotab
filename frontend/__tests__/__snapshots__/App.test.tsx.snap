// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> it matches the snapshot 1`] = `
<div
  className="ant-spin-nested-loading board-container"
>
  <div
    className="ant-spin-container"
  >
    <h1>
      Sudoku Board
    </h1>
    <div
      className="sudoku_board"
    >
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
      <div
        className="cell"
        onClick={[Function]}
        role="button"
      >
        <span />
      </div>
    </div>
    <div
      className="button-wrapper"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-lg"
        onClick={[Function]}
        type="button"
      >
        <span>
          Reload
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<App /> it matches the snapshot 2`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(App) />
</ContextProvider>
`;

exports[`<App /> render board 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(App)>
    <App
      loading={true}
      setLoading={[Function]}
    >
      <Spin
        size="default"
        spinning={true}
        wrapperClassName="board-container"
      >
        <div
          className="ant-spin-nested-loading board-container"
        >
          <div
            key="loading"
          >
            <div
              className="ant-spin ant-spin-spinning"
            >
              <span
                className="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  className="ant-spin-dot-item"
                />
                <i
                  className="ant-spin-dot-item"
                />
                <i
                  className="ant-spin-dot-item"
                />
                <i
                  className="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
          <div
            className="ant-spin-container ant-spin-blur"
            key="container"
          >
            <h1>
              Sudoku Board
            </h1>
            <Sudoku
              onChange={[Function]}
              puzzleArray={
                Array [
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                ]
              }
              selected={Array []}
            >
              <div
                className="sudoku_board"
              >
                <div
                  className="cell"
                  key="0"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="1"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="2"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="3"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="4"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="5"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="6"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="7"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="8"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="9"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="10"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="11"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="12"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="13"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="14"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="15"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="16"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="17"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="18"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="19"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="20"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="21"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="22"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="23"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="24"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="25"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="26"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="27"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="28"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="29"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="30"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="31"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="32"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="33"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="34"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="35"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="36"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="37"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="38"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="39"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="40"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="41"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="42"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="43"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="44"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="45"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="46"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="47"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="48"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="49"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="50"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="51"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="52"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="53"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="54"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="55"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="56"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="57"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="58"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="59"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="60"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="61"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="62"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="63"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="64"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="65"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="66"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="67"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="68"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="69"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="70"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="71"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="72"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="73"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="74"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="75"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="76"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="77"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="78"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="79"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
                <div
                  className="cell"
                  key="80"
                  onClick={[Function]}
                  role="button"
                >
                  <span />
                </div>
              </div>
            </Sudoku>
            <div
              className="button-wrapper"
            >
              <Button
                block={false}
                ghost={false}
                htmlType="button"
                loading={false}
                onClick={[Function]}
                size="large"
                type="primary"
              >
                <Wave>
                  <button
                    className="ant-btn ant-btn-primary ant-btn-lg"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      key=".0"
                    >
                      Reload
                    </span>
                  </button>
                </Wave>
              </Button>
            </div>
          </div>
        </div>
      </Spin>
    </App>
  </Connect(App)>
</Provider>
`;
